core/thread/inc/ROOT/TThreadedObject.hxx

```diff
@@ -30,6 +30,18 @@ class TH1;
 
 namespace ROOT {
 
+   /**
+    * \class ROOT::TNumThreads
+    * \brief Defines the number of threads in some of ROOT's interfaces.
+    */
+   struct TNumThreads {
+      int fVal; // number of slots
+      friend bool operator==(TNumThreads lhs, TNumThreads rhs) { return lhs.fVal == rhs.fVal; }
+      friend bool operator!=(TNumThreads lhs, TNumThreads rhs) { return lhs.fVal != rhs.fVal; }
+   };
```

Comment:
> It should probably always be "slots" in the context of TThreadedObject?

When a user thinks about "how many", then the unit should threads. `->` uses thread id. So slots internally (for the vector etc), but threads externally - that's how I understood it. Would you prefer me to use the (more abstract and less user-friendly) term "slots"?

Reply:
I agree that users understand threads better than slots. The problem is that "slot" is not strictly an internal concept, not always at least. It leaks out if e.g. you have N threads but only use TThreadedObject from 2 (can happen because of manual scheduling or also with a thread pool): then you need to know that TThreadedObject will have nSlots internal copies of your object and will merge nSlots copies, even if you only filled/used 2.

Reply:
OK I trust your verdict here. I then also changed `kNumThreadsIMT` => `kIMTPoolSize`.